### PR TITLE
feat: Add pricing link option to global header

### PR DESCRIPTION
**Pull request by jakemhiller**
### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change
- [ ] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/30642 '#30642')  | [Monolith Env](https://pr-30642-monolith.dev-eks.codecademy.com/ 'Monolith PR Env') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/portal-app/pull/1948)  | [Portal Env](http://www.google.fr/ 'Named link title')   |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.124-alpha.effc85.0<br/>@codecademy/gamut-labs@31.11.13-alpha.effc85.0<br/>@codecademy/styleguide@57.1.2-alpha.effc85.0<br/></pre>

File: packages/gamut-labs/src/GlobalHeader/GlobalHeaderVariants.tsx [hunk 1]
```diff
@@ -31,6 +32,10 @@ const catalogComponent = (user?: User) =>
     ? catalogDropdown(user?.hideCareerPaths)
     : courseCatalog;
 
+// Simplify pricing dropdown to a normal link for users in India
+const pricingComponent = (user?: User) =>
+  user?.geo === 'IN' ? pricingLink : pricingDropdown;
```
Comment (jakemhiller) [inline packages/gamut-labs/src/GlobalHeader/GlobalHeaderVariants.tsx:37]: i debated making this an option that gets passed in, but i think it's going to change significantly in the next round of updates, so if i made an option that switched it between a link and a dropdown, i'd likely need to completely change that to something more India specific later 🤷

**Review by dreamwasp — APPROVED** [review]

**Review by aresnik11 — APPROVED** [review]
🚀 should we add another variation in storybook for india users?

**Comment by jakemhiller** [comment]
@aresnik11 yeah, i'll add it in

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.125-alpha.5406e1.0<br/>@codecademy/gamut-labs@31.11.14-alpha.5406e1.0<br/>@codecademy/styleguide@57.2.1-alpha.5406e1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://627ebebc4b1cc609952ae978--gamut-preview.netlify.app>https://627ebebc4b1cc609952ae978--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/627ebebc4b1cc609952ae978>Deploy Logs</a></p>

**Pull request merged**